x-pack/plugins/connectors_networking/server/docs/poc.md

```diff
@@ -0,0 +1,345 @@
+a path.  For example, you might use the default timeout for most requests,
+but then a certain API like `/api/digits-of-pi` might need a longer timeout,
+so it would be nice to constrain the longer timeout to just that URL.  Perhaps
+that's another good reason to remove the `timeout` option.  We could re-visit
+allowing path elements (and query string params? headers?) later.
+
+### CNO's are "matched" to URLs, not specifically assigned
+
+Rather than have to assign a CNO to a specific connector, we're taking
+advantage of the fact that TLS options are specific to the hostname and port,
+and only applicable to https and smtp.  This allows us to create the CNO's
+independent of the connectors - they could be created BEFORE the connectors
+are created.  And once created, and new connectors using the same hostname
+and port will pick up the CNO without any direct association.
+
+This works out great, in practice, but am wondering if I missed some gotchas
+with it.
+
+In fact, having written that bit above about creating CNO's "early", these CNO's
+are perhaps good fodder for Kibana config - a way of pre-registering them, where
+we could (potentially) have the customers provide PEMs as file names (for
+on-prem; clould would need to support PEM's as strings of the file contents (7
+bit ascii anyway!)).
+
+In fact, having written that, if we could do this, we could perhaps ship this
+without any management UI for now, and only allow CNO's via config.  We 
+wouldn't need the saved objects at all.  
+
+:wondering:
+
+### CNO ids will be canonical URLs
+
+Kinda.  We don't want `/` chars in there, because we want to use "ids" as 
+path parameters for REST APIs, so we'll create keys with protocol / hostname /
+port, separated by `:` chars.  Eg, URL `https://elastic.co` becomes key/id
+`https:elastic.co:443`.
+
+I know generally we use UUIDs as saved object ids, and especially for encrypted
+saved objects ids - is it a problem using these as ids?  
+
+The reason we want a canonical URL as the id is so that we can guarantee
+uniqueness of the documents.  If we used UUIDs as ids, and the canonical
+URL is a property we try to keep unique ... things will be painful.  We'll
+potentially have multiple CNOs with the same canonical URLs, so will have
+to do some kind of maintenance check for duplicates, and then how do we even
+resolve that?
+
+### plugin client only exposes a findByUrl() function
+
+I think that plugin clients will only need access to the `findByUrl()`
+functionality, they won't need access to the "CRUD"-level access that the
+HTTP APIs will need.  So we shouldn't expose those.  So this client is a little
+different than some of the usual alerting ones, where the HTTP request
+handlers typically get access to the plugin client to do their work. 
+
+### access to the tls.ca field (and future "secret" fields?) via HTTP
+
+Currently the only field I think we need to encrypt is the `tls.ca` field,
```

Comment:
While not recommended, someone may choose to store credentials in their url. I'm not sure if we support, prevent, or 🤷‍♂️ this today: `https://elastic:changeme@my-connector-endpoint.com`

Reply:
I was actually thinking we probably want to take their URL provided, and canonicalize it to proto-hostname-port triplet - so we'd strip those out when validating their data.